logging.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE NamedFieldPuns #-}
+
+import Control.Applicative ((<|>))
+import Control.Exception.Safe (catchAny, displayException)
+import Control.Monad (guard)
+import Data.Foldable (fold)
+import System.Directory (doesFileExist)
+import System.IO (hPutStrLn, stderr)
+
+
+data Log =
+  Log {
+    logInfo :: String -> IO (),
+    logError :: String -> IO ()
+  }
+
+
+-- A log that writes to the console.
```

Comment:
I don't think these comments are necessary, since you'll be adding your own I think.